src/extension/CMakeLists.txt

```diff
@@ -14,45 +14,192 @@ add_library(thrive_extension SHARED
   core/GodotJoltConversions.hpp
 )
 
-if(MSVC)
-  target_compile_options(thrive_extension PRIVATE /W4 /wd4068)
+target_sources(thrive_extension PRIVATE ${EXTENSION_SOURCES})
 
-  if(WARNINGS_AS_ERRORS)
-    target_compile_options(thrive_extension PRIVATE /WX)
-  endif()
+# Print files and configure build
+print_source_files(thrive_extension)
 
-  # TODO debug symbols in release for MSVC (and Distribution mode)
+# Common compile definitions for both OS
```

Comment:
Any comments etc. referring to Thrive being just for TWO OSes is not going to be good because in the not too distant future *all* of this stuff also needs to be on mac. For this reason, because I think I have to do it, I don't want this huge set of cmake changes that aren't what I'm used to.